### PR TITLE
Enabled the set collider API on an actor

### PR DESCRIPTION
**Pull request by tombuMS**
Updated the API for enabling colliders to use the actor patching mechanism instead of the separate payload for enabling a collider.

This adds two overloaded functions on the actor for setting a box or sphere collider:

`public setCollider(
        colliderType: 'sphere',
        // collisionLayer: CollisionLayer,
        isTrigger: boolean,
        center?: Vector3Like,
        radius?: number): void;`

`public setCollider(
        colliderType: 'box',
        // collisionLayer: CollisionLayer,
        isTrigger: boolean,
        center?: Vector3Like,
        size?: Vector3Like): void;`

### CODE REVIEW

File: packages/sdk/src/types/runtime/actor.ts [hunk 1]
```diff
@@ -694,4 +684,50 @@ export class Actor implements ActorLike, Patchable<ActorLike> {
...
+        }
+    }
+
+    private _setCollider(collider: Partial<ColliderLike>) {
+        if (this._collider) {
+            unobserve(this._collider);
+            this._collider = undefined;
+        }
+
+        if (!this._collider) {
```
Comment (eanders-MS) [inline packages/sdk/src/types/runtime/actor.ts]: >if (!this._collider) { [](start = 8, length = 22)

I think this check is unnecessary. this._collider will always be undefined here.

File: packages/sdk/src/types/runtime/collider.ts [hunk 2]
```diff
@@ -3,54 +3,64 @@
...
-    constructor(private $owner: Actor) { }
-
-    public copy(from: Partial<ColliderLike>): this {
-        if (!from) return this;
-        if (from.enabled !== undefined) this.enabled = from.enabled;
-        if (from.isTrigger !== undefined) this.isTrigger = from.isTrigger;
-        if (from.collisionLayer !== undefined) this.collisionLayer = from.collisionLayer;
-        if (from.colliderParams !== undefined) this._colliderParams = from.colliderParams;
-        return this;
+    constructor(private $owner: Actor, initFrom: Partial<ColliderLike>) {
```
Comment (eanders-MS) [inline packages/sdk/src/types/runtime/collider.ts]: >constructor [](start = 4, length = 11)

Do you want to mark the constructor @hidden? I think we want it excluded from documentation.

File: packages/sdk/src/types/runtime/actor.ts (hunk 1 above)
Comment (tombuMS) [reply packages/sdk/src/types/runtime/actor.ts]: Yeah.  That was a left over check from before changing the enable to set paradigm.  Removing it.

File: packages/sdk/src/types/runtime/collider.ts (hunk 2 above)
Comment (tombuMS) [reply packages/sdk/src/types/runtime/collider.ts]: Yep. Added.

**Pull request merged**